pyproject.toml

```diff
@@ -53,19 +68,123 @@ extend_exclude = ["docs"]
 requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning"]
 build-backend = "poetry_dynamic_versioning.backend"
 
-[tool.black]
-line-length = 100
-target-version = ["py39", "py310", "py311"]
-
-[tool.isort]
-profile = "black"
-multi_line_output = 3
-include_trailing_comma = true
-reverse_relative = true
 
 # Ref: https://github.com/codespell-project/codespell#using-a-config-file
 [tool.codespell]
 skip = '.git,*.pdf,*.svg,go.sum,*.lock'
 check-hidden = true
 ignore-regex = '(^\s*"image/\S+": ".*|\b(KEGG.BRITE|mor.nlm.nih.gov)\b)'
 ignore-words-list = 'infarction,amination,assertIn,brite,ehr,mor,brite,nin,mirgate'
+
+[tool.ruff]
+line-length = 100
+target-version = "py312"
+
+[tool.ruff.format]
+exclude = [
+  "docs/",
+  "src/linkml_map/datamodel",
+  "tests/input"
+]
+
+[tool.ruff.lint]
+exclude = [
+  "docs/",
+  "src/linkml_map/datamodel",
+  "tests/input"
+]
+
+select = [
```

Comment:
This list is all the linter extensions currently available for ruff. ATM most of these are active so the linter throws up a lot of errors; these will be fixed in an upcoming PR. Not all the linter extensions are useful or desired but I find it easier to have the list in the pyproject file rather than having to go to the ruff website and work out what you want / don't want there.